pkg/vgmanager/lvm.go

```diff
@@ -23,33 +23,207 @@ import (
 	"github.com/red-hat-storage/lvm-operator/pkg/internal"
 )
 
+type lvmError string
+
+func (s lvmError) Error() string { return string(s) }
+
+var (
+	ErrVolumeGroupNotFound = lvmError("volume group not found")
+)
+
 const (
-	lvmPath = "/usr/sbin/lvm"
+	lvmCmd      = "/usr/sbin/lvm"
+	vgCreateCmd = "/usr/sbin/vgcreate"
+	vgExtendCmd = "/usr/sbin/vgextend"
+	vgRemoveCmd = "/usr/sbin/vgremove"
+	pvRemoveCmd = "/usr/sbin/pvremove"
 )
 
+// vgsOutput represents the output of the `vgs --reportformat json` command
+type vgsOutput struct {
+	Report []struct {
+		Vg []struct {
+			Name string `json:"vg_name"`
+		} `json:"vg"`
+	} `json:"report"`
+}
+
+// pvsOutput represents the output of the `pvs --reportformat json` command
+type pvsOutput struct {
+	Report []struct {
+		Pv []struct {
+			Name   string `json:"pv_name"`
+			VgName string `json:"vg_name"`
+		} `json:"pv"`
+	} `json:"report"`
+}
+
 // VolumeGroup represents a volume group of linux lvm.
 type VolumeGroup struct {
+	// Name is the name of the volume group
 	Name string `json:"vg_name"`
+
+	// PVs is the list of physical volumes associated with the volume group
+	PVs []string `json:"pvs"`
 }
 
-// ListVolumeGroups lists all volume groups.
-func ListVolumeGroups(exec internal.Executor) ([]VolumeGroup, error) {
-	// volumeGroupMap type is used for marshalling the `vgs --reportformt json` command output
-	// Sample output:
-	// `{"report": [ {"vg": [{"vg_name":"fedora", "pv_count":"1", "lv_count":"3", "snap_count":"0", "vg_attr":"wz--n-", "vg_size":"<475.94g", "vg_free":"0 "}]}]`
-	volumeGroupMap := map[string][]map[string][]VolumeGroup{}
-	arg := []string{
+// Create creates a new volume group
+func (vg VolumeGroup) Create(exec internal.Executor, pvs []string) error {
```

Comment:
Create, extend and delete methods only returns error...and in case of success we are losing the output , which can be useful in logs just to trace what is happening. Maybe it could be a good idea to return the error and also and output string to provide the output produced by the vg command when it is executed succesfully.

Reply:
I don't think it will be very useful.  We would only be interested to know whether vg got created or not. If not, then what's the error.